src/formats/png/reader.zig

```diff
@@ -0,0 +1,1041 @@
+                    data.dest_row[pixel_pos + 3] = entry.a;
+                }
+            },
+            .indexed16 => {
+                while (pixel_pos + 3 < data.dest_row.len) : (pixel_pos += pixel_stride) {
+                    //const index_buf: [2]u8 = .{data.dest_row[pixel_pos], data.dest_row[pixel_pos + 1]};
+                    const index = std.mem.bytesToValue(u16, &[2]u8{ data.dest_row[pixel_pos], data.dest_row[pixel_pos + 1] });
+                    const entry = self.palette[index];
+                    data.dest_row[pixel_pos] = entry.r;
+                    data.dest_row[pixel_pos + 1] = entry.g;
+                    data.dest_row[pixel_pos + 2] = entry.b;
+                    data.dest_row[pixel_pos + 3] = entry.a;
+                }
+            },
+            else => unreachable,
+        }
+
+        return .rgba32;
+    }
+};
+
+/// The options you need to pass to PNG reader. If you want default options
+/// with buffer for temporary allocations on the stack and default set of
+/// processors just use this:
+/// var def_options = DefOptions{};
+/// png.reader.load(main_allocator, def_options.get());
+/// Note that application can define its own DefPngOptions in the root file
+/// and all the code that uses DefOptions will actually use that.
+pub const ReaderOptions = struct {
+    /// Allocator for temporary allocations. The constant required_temp_bytes defines
+    /// the maximum bytes that will be allocated from it. Some temp allocations depend
+    /// on the image size so they will use the main allocator since we can't guarantee
+    /// they are bounded. They will be allocated after the destination image to
+    /// reduce memory fragmentation and freed internally.
+    temp_allocator: Allocator,
+
+    /// Default is no processors so they are not even compiled in if not used.
+    /// If you want a default set of processors create a DefProcessors object
+    /// call get() on it and pass that here.
+    /// Note that application can define its own DefPngProcessors and all the
+    /// code that uses DefProcessors will actually use that.
+    processors: []ReaderProcessor = &[_]ReaderProcessor{},
+
+    pub fn init(temp_allocator: Allocator) ReaderOptions {
+        return .{ .temp_allocator = temp_allocator };
+    }
+
+    pub fn initWithProcessors(temp_allocator: Allocator, processors: []ReaderProcessor) ReaderOptions {
+        return .{ .temp_allocator = temp_allocator, .processors = processors };
+    }
+};
+
+// decompressor.zig:294 claims to use up to 300KiB from provided allocator but when
+// testing with huge png file it used 760KiB.
+// Original zlib claims it only needs 44KiB so next task is to rewrite zig's zlib :).
+pub const required_temp_bytes = 800 * 1024;
+
+const root = @import("root");
+
+/// Applications can override this by defining DefPngProcessors struct in their root source file.
+pub const DefProcessors = if (@hasDecl(root, "DefPngProcessors"))
```

Comment:
`DefPngProcessors` -> `DefaultPngProcessors`

Reply:
Should we put `Png` into these type names considering it will be mostly used like this:
```
const png = @import("formats/png");
var processors = png.DefaultProcessors;
```
and the `png` namespace already tells you it is for png? If you want to import that specific symbol into your namespace you can do `const DefaultPngProcessors = png.DefaultProcessors;`.

Reply:
Remove the Png prefix from those types